pkg/csi/service/wcpguest/controller.go

```diff
@@ -420,6 +456,137 @@ func (c *controller) ControllerPublishVolume(ctx context.Context, req *csi.Contr
+		log.Error(msg)
+		return nil, status.Errorf(codes.Internal, msg)
+	}
+	// Attacher timeout, default is set to 4 minutes
+	timeoutSeconds := int64(getAttacherTimeoutInMin(ctx) * 60)
+	// Adding watch on the CnsFileAccessConfig instance to register for updates
+	watchCnsFileAccessConfig, err := cnsFileAccessConfigWatcher.Watch(metav1.ListOptions{
+		FieldSelector:   fields.SelectorFromSet(fields.Set{"metadata.name": cnsFileAccessConfigInstance.Name}).String(),
+		ResourceVersion: cnsFileAccessConfigInstance.ResourceVersion,
+		TimeoutSeconds:  &timeoutSeconds,
+	})
+	if err != nil {
+		msg := fmt.Sprintf("failed to watch cnsfileaccessconfig %q with Error: %v", cnsFileAccessConfigInstance.Name, err)
+		log.Error(msg)
+		return nil, status.Errorf(codes.Internal, msg)
+	}
+	defer watchCnsFileAccessConfig.Stop()
+	var cnsFileAccessConfigInstanceErr string
+	// Watch all update events made on CnsFileAccessConfig instance until accessPoints is set
+	for {
+		log.Debugf("Waiting for update on cnsfileaccessconfigs: %q", cnsFileAccessConfigInstance.Name)
+		event := <-watchCnsFileAccessConfig.ResultChan()
+		cnsfileaccessconfig, ok := event.Object.(*cnsfileaccessconfigv1alpha1.CnsFileAccessConfig)
+		if !ok {
+			msg := fmt.Sprintf("Watch on cnsfileaccessconfig instance %q timed out. Last seen error on the instance=%q", cnsFileAccessConfigInstance.Name, cnsFileAccessConfigInstanceErr)
+			log.Error(msg)
+			return nil, status.Errorf(codes.Internal, msg)
+		}
+		if cnsfileaccessconfig.Name != cnsFileAccessConfigInstanceName {
+			log.Debugf("Observed cnsFileAccessConfig instance name: %q, expecting cnsFileAccessConfig instance name: %q. Continuing...", cnsfileaccessconfig.Name, cnsFileAccessConfigInstanceName)
+			continue
+		}
+		// Check if SV PVC Name match with VolumeId from the request
+		if cnsfileaccessconfig.Spec.PvcName != req.VolumeId {
+			log.Debugf("Observed SV PVC Name: %q, expecting SV PVC Name: %q. Continuing...", cnsfileaccessconfig.Spec.PvcName, req.VolumeId)
+			continue
+		}
+		// Check if VM name in the cnsfileaccessconfig instance match with NodeId from the request
+		if cnsfileaccessconfig.Spec.VMName != req.NodeId {
+			log.Debugf("Observed vm name: %q, expecting vm name: %q. Continuing...", cnsfileaccessconfig.Spec.VMName, req.NodeId)
+			continue
+		}
+		log.Debugf("Observed an update on cnsfileaccessconfig: %+v", cnsfileaccessconfig)
+		if cnsfileaccessconfig.Status.Done && cnsfileaccessconfig.Status.Error == "" {
+			// Check if the updated instance has the AccessPoints
+			for key, value := range cnsfileaccessconfig.Status.AccessPoints {
+				if key == common.Nfsv4AccessPointKey {
+					publishInfo[common.AttributeDiskType] = common.DiskTypeFileVolume
+					publishInfo[common.Nfsv4AccessPoint] = value
+					break
+				}
+			}
+			if _, ok := publishInfo[common.Nfsv4AccessPoint]; ok {
+				log.Debugf("Found Nfsv4AccessPoint in publishInfo. publishInfo=%+v", publishInfo)
+				break
```

Comment:
Nit: Maybe you should return response from here, its more clear that way instead of adding a break here and then returning the response outside the for loop. This is subjective, so feel free to ignore this comment.

Reply:
+1

Reply:
Ignoring this as the behavior is not affected. Thanks for suggesting though